include/mqtt/message.hpp

```diff
@@ -529,46 +529,48 @@ class basic_publish_message {
         }
     }
 
-    template <typename Iterator>
-    basic_publish_message(Iterator b, Iterator e) {
-        if (b + 1 > e) throw remaining_length_error();
-        fixed_header_ = static_cast<std::uint8_t>(*b);
+    basic_publish_message(mqtt::string_view buf) {
+        if (buf.empty())  throw remaining_length_error();
+        fixed_header_ = static_cast<std::uint8_t>(buf.front());
         auto qos = publish::get_qos(fixed_header_);
-        ++b;
+        buf = std::move(buf).substr(1);
```

Comment:
In this case, `mqtt::string_view` is better because `basic_publish_message` doesn't store substring of `mqtt::buffer`. If we would go the way #338 in the future, then I would pass `mqtt::buffer`.

So far, I remove `std::move()`.

Reply:
Yes, i agree.

> In this case, mqtt::string_view is better because basic_publish_message doesn't store substring of mqtt::buffer. 

----------------------------------

> If we would go the way #338 in the future, then I would pass mqtt::buffer.

unless you have basic publish message hold a reference count to the shared pointer, i think std::string_view is still fine.